cfme/tests/infrastructure/test_individual_host_creds.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.flash as flash
+import pytest
+import random
+import utils.conf as conf
+
+from cfme.infrastructure.host import credential_form
+from cfme.infrastructure.provider import details_page
+from cfme.web_ui import fill
+from utils import testgen
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = testgen.provider_by_type(metafunc, ['virtualcenter', 'rhevm'])
+    testgen.parametrize(metafunc, argnames, argvalues, ids=idlist, scope="module")
+
+
+# Tests to automate BZ 1278904
+def test_single_host_bad_creds(setup_provider, provider):
+    """  Tests multiple host credentialing  with good credentials """
+
+    sel.force_navigate('infrastructure_provider', context={'provider': provider})
+    sel.click(details_page.infoblock.element("Relationships", "Hosts"))
+
+    host_name = random.choice(provider.get_yaml_data()["hosts"])
+    pytest.sel.force_navigate('infrastructure_host_edit', context={
+        'host': host_name, })
+    creds = conf.credentials['bad_credentials']
+    fill(credential_form, {'default_principal': creds['username'],
+                           'default_secret': creds['password'],
+                           'default_verify_secret': creds['password']})
+
+    sel.click(credential_form.validate_btn)
+
+    if provider.type == 'virtualcenter':
```

Comment:
Let's turn these into a dict
```python
msgs = {
    'virtualcenter': 'Cannot complete...............'
    'rhevm': 'Login failed.........'
}
flash.assert_message_match(msgs[provider.type])
```
Though, again there is, at least for ```provider.create()``` a param falled ```validate``` which I think performs the validation before it saves. This again a) should probably be used here, and b) is an argument for using the Host object.